java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/binder/ColumnBinderArrowTypeVisitor.java

```diff
@@ -0,0 +1,263 @@
+import org.apache.arrow.vector.Float4Vector;
+import org.apache.arrow.vector.Float8Vector;
+import org.apache.arrow.vector.IntVector;
+import org.apache.arrow.vector.LargeVarBinaryVector;
+import org.apache.arrow.vector.LargeVarCharVector;
+import org.apache.arrow.vector.SmallIntVector;
+import org.apache.arrow.vector.TimeMicroVector;
+import org.apache.arrow.vector.TimeMilliVector;
+import org.apache.arrow.vector.TimeNanoVector;
+import org.apache.arrow.vector.TimeSecVector;
+import org.apache.arrow.vector.TimeStampVector;
+import org.apache.arrow.vector.TinyIntVector;
+import org.apache.arrow.vector.VarBinaryVector;
+import org.apache.arrow.vector.VarCharVector;
+import org.apache.arrow.vector.types.pojo.ArrowType;
+
+/**
+ * Visitor to create the base ColumnBinder for a vector.
+ * <p>
+ * To handle null values, wrap the returned binder in a {@link NullableColumnBinder}.
+ */
+public class ColumnBinderArrowTypeVisitor implements ArrowType.ArrowTypeVisitor<ColumnBinder> {
+  private final FieldVector vector;
+  private final Integer jdbcType;
+
+  /**
+   * Create a binder using a custom JDBC type code.
+   *
+   * @param vector The vector that the binder will wrap.
+   * @param jdbcType The JDBC type code (or null to use the default).
+   */
+  public ColumnBinderArrowTypeVisitor(FieldVector vector, Integer jdbcType) {
+    this.vector = vector;
+    this.jdbcType = jdbcType;
+  }
+
+  @Override
+  public ColumnBinder visit(ArrowType.Null type) {
+    throw new UnsupportedOperationException("No column binder implemented for type " + type);
+  }
+
+  @Override
+  public ColumnBinder visit(ArrowType.Struct type) {
+    throw new UnsupportedOperationException("No column binder implemented for type " + type);
+  }
+
+  @Override
+  public ColumnBinder visit(ArrowType.List type) {
+    throw new UnsupportedOperationException("No column binder implemented for type " + type);
+  }
+
+  @Override
+  public ColumnBinder visit(ArrowType.LargeList type) {
+    throw new UnsupportedOperationException("No column binder implemented for type " + type);
+  }
+
+  @Override
+  public ColumnBinder visit(ArrowType.FixedSizeList type) {
+    throw new UnsupportedOperationException("No column binder implemented for type " + type);
+  }
+
+  @Override
+  public ColumnBinder visit(ArrowType.Union type) {
+    throw new UnsupportedOperationException("No column binder implemented for type " + type);
+  }
+
+  @Override
+  public ColumnBinder visit(ArrowType.Map type) {
+    throw new UnsupportedOperationException("No column binder implemented for type " + type);
+  }
+
+  @Override
+  public ColumnBinder visit(ArrowType.Int type) {
+    switch (type.getBitWidth()) {
+      case 8:
+        if (type.getIsSigned()) {
+          return jdbcType == null ? new TinyIntBinder((TinyIntVector) vector) :
+              new TinyIntBinder((TinyIntVector) vector, jdbcType);
+        } else {
+          throw new UnsupportedOperationException(
```

Comment:
Maybe we can extract this statement for all type widths, at the beginning of this method?

Reply:
Done.